File: eio2024-lv/2024-10-07-lv/ekraan/templates/template.cpp
#include <iostream>
using namespace std;

int main() {
	// loeme indikaatorite arvu
	int n;
	cin >> n;

	// lahendame kõik testid
	for (;;) {
		long long code;
		cin >> code;

		// kui rohkem teste pole, lõpetame
		if (code == -1) {
			break;
		}

		// TODO: küsime lisainfot, kuni leiame selle testi vastuse
		for (int i = 0; i < 4; ++i) {
			cout << "NEXT" << endl;
			cin >> code;
		}

		// TODO: väljastame vastuse ja liigume järgmise testi juurde
		cout << 42 << endl;
	}
}

File: eio2024-lv/2024-10-07-lv/ekraan/templates/template.py
# loeme indikaatorite arvu
n = int(input())

# lahendame kõik testid
while True:
    code = int(input())

    # kui rohkem teste pole, lõpetame
    if code == -1:
        break

    # TODO: küsime lisainfot, kuni leiame selle testi vastuse
    for i in range(4):
        print("NEXT", flush=True)
        code = int(input())
   
    # TODO: väljastame vastuse ja liigume järgmise testi juurde
    print(42, flush=True)
